sale_outplacement/models/sale.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: UTF-8 -*-
+
+###############################################################################
+#                                                                             #
+#    OpenERP, Open Source Management Solution                                 #
+#    Copyright (C) 2019 N-Development (<https://n-development.com>).          #
+#                                                                             #
+#    This program is free software: you can redistribute it and/or modify     #
+#    it under the terms of the GNU Affero General Public License as           #
+#    published by the Free Software Foundation, either version 3 of the       #
+#    License, or (at your option) any later version.                          #
+#                                                                             #
+#    This program is distributed in the hope that it will be useful,          #
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of           #
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            #
+#    GNU Affero General Public License for more details.                      #
+#                                                                             #
+#    You should have received a copy of the GNU Affero General Public License #
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.    #
+#                                                                             #
+###############################################################################
+
+
+import datetime  # Used in test
+import random  # Used in test
+import string  # Used in test
+
+from odoo import api, fields, models, _ 
+
+
+import logging
+_logger = logging.getLogger(__name__)
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.one
+    def update_order_from_ordertjansten(self):
+        res = self.env['ipf.showorder.client.config'].get_order_id(self.name)
+        # ~ res = self.env['sale.order'].get_order_from_ordertjansten()
+        for artikel in res.get('artikelList',[]):
+            if not artikel['tlrId'] in self.order_line.mapped('product_id.tlr_ref'):
+                product = self.env['product.product'].search([('tlr_ref','=',artikel['tlrId'])],limit=1)
+                if not product:
+                    product = self.env['product.product'].create({
+                        "tlr_ref": artikel["tlrId"],
+                        "name":  artikel["namn"],
+                        "list_price": artikel["nuvarandeAPris"],
+      
+                    })
+                self.env['sale.order.line'].create({
+                    'order_id': self.id,
+                    'product_id': product.id,
+                    'qty': int(artikel['nuvarandeAntal']),
+                    'forvantatantal': int(artikel['forvantatAntal']),
+                    })
+    @api.model
+    def get_order_from_ordertjansten(self):
```

Comment:
this function returns hardcoded test data, maybe rename it to get_order_from_ordertjansten_test

Reply:
Good idea, that means that we ahve to reame all references (i think one server action) too and test it again.